src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -2176,6 +2179,49 @@ new_fxstat(int vers, int fd, struct stat *buf)
 	return 0;
 }
 
+int
+fstat(int fd, struct stat *buf)
```

Comment:
Is this part of the same fix?

It seems likely to me that we'd want fstatat on the 2.4 branch but not el9 support so perhaps this would be better as two PRs?

Reply:
fstat and fstatat are added in the newer glibc (2.34) on EL9. "fxstat" and "fxstatat" are provided in the glibc (2.28) on el8. Both of them are intercepted in 2.4.